diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index e1da11405c1f6..74a22bc4399d3 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -529,10 +529,10 @@ config("compiler") {
   # Linux/Android/Fuchsia common flags setup.
   # ---------------------------------
   if (is_linux || is_chromeos || is_android || is_fuchsia) {
-    asmflags += [ "-fPIC" ]
-    cflags += [ "-fPIC" ]
-    ldflags += [ "-fPIC" ]
-    rustflags += [ "-Crelocation-model=pic" ]
+    asmflags += [ "-fPIC", "-w", ]
+    cflags += [ "-fPIC", "-w", ]
+    ldflags += [ "-fPIC", "-w", ]
+    rustflags += [ "-Crelocation-model=pic", "-Awarnings", ]
 
     if (!is_clang) {
       # Use pipes for communicating between sub-processes. Faster.
@@ -653,7 +653,7 @@ config("compiler") {
     if (is_win) {
       cflags += [ "/clang:-ffp-contract=off" ]
     } else {
-      cflags += [ "-ffp-contract=off" ]
+      cflags += [ "-ffp-contract=fast" ]
     }
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
@@ -980,6 +980,11 @@ config("compiler") {
 
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
+
+    "-Ctarget-cpu=x86-64-v3",
+    "-Cllvm-args=-fp-contract=fast",
+    "-Ztune-cpu=skylake",
+    "-Ctarget-feature=-fast-gather,+prefer-no-gather,+prefer-no-scatter",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1260,12 +1265,23 @@ config("disallow_unstable_features") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    if (thin_lto_enable_optimizations) {
+      lto_opt_level = 2
+    } else {
+      lto_opt_level = 0
+    }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-adce-remove-loops=true",
+        "-Wl,-mllvm,-x86-align-branch-boundary=32",
+        "-Wl,-mllvm,-x86-align-branch=jcc+jmp+indirect",
+        "-Wl,-z,pack-relative-relocs",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1301,6 +1317,21 @@ config("thinlto_optimize_max") {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-import-hot-multiplier=15",
+        "-Wl,-mllvm,-adce-remove-loops=true",
+        "-Wl,-mllvm,-enable-gvn-hoist",
+        "-Wl,-mllvm,-enable-gvn-sink",
+        "-Wl,-mllvm,-machine-sink-split",
+        "-Wl,-mllvm,-phi-node-folding-threshold=4",
+        "-Wl,-mllvm,-sink-insts-to-avoid-spills",
+        "-Wl,-mllvm,-spec-exec-max-speculation-cost=15",
+        "-Wl,-mllvm,-x86-align-branch-boundary=32",
+        "-Wl,-mllvm,-x86-align-branch=jcc+jmp+indirect",
+        "-Wl,-mllvm,-x86-cmov-converter",
+        "-Wl,-z,pack-relative-relocs",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1308,7 +1339,7 @@ config("thinlto_optimize_max") {
       # Rust to preemptively do so during compilation too or they conflict. But
       # we do want Rust to generate LTO metadata in order for the linker to do
       # its job.
-      rustflags = [ "-Clinker-plugin-lto=yes" ]
+      rustflags = [ "-Clinker-plugin-lto=yes", ]
     } else {
       # Don't include bitcode if it won't be used.
       rustflags = [ "-Cembed-bitcode=no" ]
@@ -1723,7 +1754,7 @@ config("runtime_library") {
   }
 
   if (use_safe_libcxx) {
-    defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE" ]
+    defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST" ]
   } else {
     defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE" ]
   }
@@ -2645,7 +2676,7 @@ config("optimize_max") {
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O3" ] + common_optimize_on_cflags
   }
   rustflags = [ "-Copt-level=3" ]
 }
diff --git a/build/config/compiler_cpu_abi.gn b/build/config/compiler_cpu_abi.gn
index ae62bab587e7d..138bb533ad6de 100644
--- a/build/config/compiler_cpu_abi.gn
+++ b/build/config/compiler_cpu_abi.gn
@@ -55,7 +55,13 @@ if ((is_posix && !is_apple) || is_fuchsia) {
       cpu_abi_cflags += [ "-m64" ]
       cpu_abi_ldflags += [ "-m64" ]
     }
-    cpu_abi_cflags += [ "-msse3" ]
+    cpu_abi_cflags += [
+      "-march=x86-64-v3",
+      "-mtune=skylake",
+      "-mno-gather",
+      "-mno-scatter",
+      "-mno-bmi2",
+    ]
     if (target_os == "chromeos" && cros_target_cpu_arch != "" &&
         is_a_target_toolchain) {
       cpu_abi_cflags += [ "-march=$cros_target_cpu_arch" ]
diff --git a/chrome/common/chrome_paths_linux.cc b/chrome/common/chrome_paths_linux.cc
index 5575c049b222f..8be111ca9ed9c 100644
--- a/chrome/common/chrome_paths_linux.cc
+++ b/chrome/common/chrome_paths_linux.cc
@@ -95,7 +95,7 @@ bool GetDefaultUserDataDirectory(base::FilePath* result) {
 #elif BUILDFLAG(GOOGLE_CHROME_BRANDING)
   std::string data_dir_basename = "google-chrome";
 #else
-  std::string data_dir_basename = "chromium";
+  std::string data_dir_basename = "chromium-clang";
 #endif
   *result = config_dir.Append(data_dir_basename + GetChannelSuffixForDataDir());
   return true;
diff --git a/chrome/installer/linux/common/repo.cron b/chrome/installer/linux/common/repo.cron
index 571fb836f3ac1..e69de29bb2d1d 100755
--- a/chrome/installer/linux/common/repo.cron
+++ b/chrome/installer/linux/common/repo.cron
@@ -1,37 +0,0 @@
-#!/bin/sh
-#
-# Copyright 2009 The Chromium Authors
-# Use of this source code is governed by a BSD-style license that can be
-# found in the LICENSE file.
-#
-# This script is part of the @@PACKAGE package.
-#
-# It creates the repository configuration file for package updates, and it
-# monitors that config to see if it has been disabled by the overly aggressive
-# distro upgrade process (e.g.  intrepid -> jaunty). When this situation is
-# detected, the respository will be re-enabled. If the respository is disabled
-# for any other reason, this won't re-enable it.
-#
-# This functionality can be controlled by creating the $DEFAULTS_FILE and
-# setting "repo_reenable_on_distupgrade" to "true" or "false" as desired.
-# An empty $DEFAULTS_FILE is the same as setting it to "false".
-#
-# The $DEFAULTS_FILE also has a setting "repo_add_once" which can be set
-# to "false" to prevent the package install from adding the repo altogether.
-
-@@include@@../common/variables.include
-
-@@include@@apt.include
-
-## MAIN ##
-if [ -r "$DEFAULTS_FILE" ]; then
-  . "$DEFAULTS_FILE"
-fi
-
-install_key
-
-if [ "$repo_add_once" = "true" ]; then
-  create_sources_lists
-elif [ "$repo_reenable_on_distupgrade" = "true" ]; then
-  install_deb822_sources
-fi
diff --git a/chrome/installer/linux/common/rpmrepo.cron b/chrome/installer/linux/common/rpmrepo.cron
index 90ff896ea5ede..e69de29bb2d1d 100755
--- a/chrome/installer/linux/common/rpmrepo.cron
+++ b/chrome/installer/linux/common/rpmrepo.cron
@@ -1,59 +0,0 @@
-#!/bin/sh
-#
-# Copyright 2009 The Chromium Authors
-# Use of this source code is governed by a BSD-style license that can be
-# found in the LICENSE file.
-#
-# This script is part of the @@PACKAGE package.
-#
-# It creates the repository configuration file for package updates, since
-# we cannot do this during the @@PACKAGE installation since the repository
-# is locked.
-#
-# This functionality can be controlled by creating the $DEFAULTS_FILE and
-# setting "repo_add_once" to "true" or "false" as desired. An empty
-# $DEFAULTS_FILE is the same as setting the value to "false".
-
-@@include@@../common/variables.include
-
-@@include@@rpm.include
-
-## MAIN ##
-if [ -r "$DEFAULTS_FILE" ]; then
-  . "$DEFAULTS_FILE"
-fi
-
-install_rpm_key
-
-if [ "$repo_add_once" = "true" ]; then
-  determine_rpm_package_manager
-
-  # Let's run through this for each supported package manager as detected...
-  for PACKAGEMANAGER in ${PACKAGEMANAGERS[*]}
-  do
-    # The initial install happens in the post-install scripts, but there have been
-    # reports of configuration problems, so just verify that everything looks
-    # good, and if not, try to install again.
-    verify_install $PACKAGEMANAGER
-    if [ $? -ne 0 ]; then
-      install_${PACKAGEMANAGER}
-    fi
-  done
-
-  if [ $? -eq 0 ]; then
-    # Do this for each supported package manager...
-    for PACKAGEMANAGER in ${PACKAGEMANAGERS[*]}
-    do
-      # Before we quit auto-configuration, check that everything looks sane, since
-      # part of this happened during package install and we don't have the return
-      # value of that process.
-      verify_install $PACKAGEMANAGER
-      if [ $? -eq 0 ]; then
-        sed -i -e 's/[[:space:]]*repo_add_once=.*/repo_add_once="false"/' \
-          "$DEFAULTS_FILE"
-      fi
-    done
-  fi
-else
-  update_bad_repo
-fi
diff --git a/chrome/installer/linux/debian/build.py b/chrome/installer/linux/debian/build.py
index 2d023ed81b760..9ab90487cd1e3 100755
--- a/chrome/installer/linux/debian/build.py
+++ b/chrome/installer/linux/debian/build.py
@@ -209,9 +209,9 @@ def main() -> None:
             installer.run_command(["ar", "-x", pkg_basename], cwd=tmp_file_dir)
             installer.run_command([
                 "xz",
-                "-z9",
+                "-z6",
+                "-e",
                 f"-T{args.xz_nthreads}",
-                "--lzma2=dict=256MiB",
                 str(tmp_file_dir / "data.tar"),
             ])
             installer.run_command(
diff --git a/chrome/installer/linux/debian/postrm b/chrome/installer/linux/debian/postrm
index 3e1cadc8296ca..0cb44dbf7ad77 100755
--- a/chrome/installer/linux/debian/postrm
+++ b/chrome/installer/linux/debian/postrm
@@ -20,17 +20,3 @@ fi
 @@include@@../common/symlinks.include
 
 remove_udev_symlinks
-
-# Only remove the defaults file if it is not empty. An empty file was probably
-# put there by the sysadmin to disable automatic repository configuration, as
-# per the instructions on the package download page.
-if [ -s "$DEFAULTS_FILE" ]; then
-  # Make sure the package defaults are removed before the repository config,
-  # otherwise it could result in the repository config being removed, but the
-  # package defaults remain and are set to not recreate the repository config.
-  # In that case, future installs won't recreate it and won't get auto-updated.
-  rm "$DEFAULTS_FILE" || exit 1
-fi
-# Remove any Google repository added by the package.
-clean_sources_lists
-uninstall_key
diff --git a/chrome/installer/linux/rpm/build.py b/chrome/installer/linux/rpm/build.py
index 5cced7e815d64..3c422a75b4527 100755
--- a/chrome/installer/linux/rpm/build.py
+++ b/chrome/installer/linux/rpm/build.py
@@ -132,7 +132,7 @@ def main() -> None:
         (rpm_build_dir / "RPMS").mkdir(parents=True, exist_ok=True)
 
         if args.official:
-            compression_opt = "_binary_payload w9.xzdio"
+            compression_opt = "_binary_payload w6T0.xzdio"
         else:
             compression_opt = "_binary_payload w0.gzdio"
 
diff --git a/content/common/gpu_pre_sandbox_hook_linux.cc b/content/common/gpu_pre_sandbox_hook_linux.cc
index 2e53794fa3374..1786fda6f463d 100644
--- a/content/common/gpu_pre_sandbox_hook_linux.cc
+++ b/content/common/gpu_pre_sandbox_hook_linux.cc
@@ -262,6 +262,18 @@ void AddAmdGpuPermissions(std::vector<BrokerFilePermission>* permissions) {
       "/usr/lib64/dri/r300_dri.so",
       "/usr/lib64/dri/r600_dri.so",
       "/usr/lib64/dri/radeonsi_dri.so",
+      // GPU Log Warning Workaround
+      "/usr/lib/libEGL.so.1",
+      "/usr/lib/libGLESv2.so.2",
+      "/usr/lib/libglapi.so.0",
+      "/usr/lib/dri/r300_dri.so",
+      "/usr/lib/dri/r600_dri.so",
+      "/usr/lib/dri/radeonsi_dri.so",
+      "/usr/lib/libvulkan_radeon.so",
+      "/usr/share/vulkan/icd.d",
+      "/usr/share/vulkan/icd.d/radeon_icd.x86_64.json",
+      "/etc/vulkan/icd.d",
+      "/etc/vulkan/icd.d/radeon_icd.x86_64.json",
       // Allow libglvnd files and libs.
       "/usr/share/glvnd/egl_vendor.d",
       "/usr/share/glvnd/egl_vendor.d/50_mesa.json",
diff --git a/media/base/supported_types.cc b/media/base/supported_types.cc
index c4d25f792f306..f882457124e59 100644
--- a/media/base/supported_types.cc
+++ b/media/base/supported_types.cc
@@ -26,7 +26,6 @@
 #endif
 
 #if BUILDFLAG(IS_ANDROID)
-
 // TODO(dalecurtis): This include is not allowed by media/base since
 // media/base/android is technically a different component. We should move
 // supported_types*.{cc,h} out of media/base to fix this.
@@ -39,596 +38,587 @@
 
 namespace media {
 
-namespace {
+  namespace {
+
+    template <typename T>
+    class SupplementalProfileCache {
+    public:
+      void UpdateCache(const base::flat_set<T>& profiles) {
+        base::AutoLock lock(profiles_lock_);
+        profiles_ = profiles;
+      }
+      bool IsProfileSupported(T profile) {
+        base::AutoLock lock(profiles_lock_);
+        return profiles_.find(profile) != profiles_.end();
+      }
+
+    private:
+      base::Lock profiles_lock_;
+      base::flat_set<T> profiles_ GUARDED_BY(profiles_lock_);
+    };
+
+    SupplementalProfileCache<VideoCodecProfile>*
+    GetSupplementalDecoderVideoProfileCache() {
+      static base::NoDestructor<SupplementalProfileCache<VideoCodecProfile>> cache;
+      return cache.get();
+    }
+
+    SupplementalProfileCache<AudioType>* GetSupplementalDecoderAudioTypeCache() {
+      static base::NoDestructor<SupplementalProfileCache<AudioType>> cache;
+      return cache.get();
+    }
+
+    SupplementalProfileCache<VideoCodecProfile>*
+    GetSupplementalEncoderVideoProfileCache() {
+      static base::NoDestructor<SupplementalProfileCache<VideoCodecProfile>> cache;
+      return cache.get();
+    }
+
+    bool IsDecoderSupportedHdrMetadata(const VideoType& type) {
+      switch (type.hdr_metadata_type) {
+        case gfx::HdrMetadataType::kNone:
+          return true;
+
+        case gfx::HdrMetadataType::kSmpteSt2086:
+          // HDR metadata is currently only used with the PQ transfer function.
+          // See gfx::ColorTransform for more details.
+          return type.color_space.transfer() ==
+          VideoColorSpace::TransferID::SMPTEST2084;
+
+          // 2094-10 SEI metadata is not the same as Dolby Vision RPU metadata, Dolby
+          // Vision decoders on each platform only support Dolby Vision RPU metadata.
+        case gfx::HdrMetadataType::kSmpteSt2094_10:
+        case gfx::HdrMetadataType::kSmpteSt2094_40:
+          return false;
+      }
+    }
+
+    bool IsColorSpaceSupported(const VideoColorSpace& color_space) {
+      switch (color_space.primaries()) {
+        // Transfers supported before color management.
+        case VideoColorSpace::PrimaryID::BT709:
+        case VideoColorSpace::PrimaryID::UNSPECIFIED:
+        case VideoColorSpace::PrimaryID::BT470M:
+        case VideoColorSpace::PrimaryID::BT470BG:
+        case VideoColorSpace::PrimaryID::SMPTE170M:
+          break;
+
+          // Supported with color management.
+        case VideoColorSpace::PrimaryID::SMPTE240M:
+        case VideoColorSpace::PrimaryID::FILM:
+        case VideoColorSpace::PrimaryID::BT2020:
+        case VideoColorSpace::PrimaryID::SMPTEST428_1:
+        case VideoColorSpace::PrimaryID::SMPTEST431_2:
+        case VideoColorSpace::PrimaryID::SMPTEST432_1:
+        case VideoColorSpace::PrimaryID::EBU_3213_E:
+          break;
+
+        case VideoColorSpace::PrimaryID::INVALID:
+          return false;
+      }
+
+      switch (color_space.transfer()) {
+        // Transfers supported before color management.
+        case VideoColorSpace::TransferID::UNSPECIFIED:
+        case VideoColorSpace::TransferID::GAMMA22:
+        case VideoColorSpace::TransferID::BT709:
+        case VideoColorSpace::TransferID::SMPTE170M:
+        case VideoColorSpace::TransferID::BT2020_10:
+        case VideoColorSpace::TransferID::BT2020_12:
+        case VideoColorSpace::TransferID::IEC61966_2_1:
+          break;
+
+          // Supported with color management.
+        case VideoColorSpace::TransferID::GAMMA28:
+        case VideoColorSpace::TransferID::SMPTE240M:
+        case VideoColorSpace::TransferID::LINEAR:
+        case VideoColorSpace::TransferID::LOG:
+        case VideoColorSpace::TransferID::LOG_SQRT:
+        case VideoColorSpace::TransferID::BT1361_ECG:
+        case VideoColorSpace::TransferID::SMPTEST2084:
+        case VideoColorSpace::TransferID::IEC61966_2_4:
+        case VideoColorSpace::TransferID::SMPTEST428_1:
+        case VideoColorSpace::TransferID::ARIB_STD_B67:
+          break;
+
+          // Never supported.
+        case VideoColorSpace::TransferID::INVALID:
+          return false;
+      }
+
+      switch (color_space.matrix()) {
+        // Supported before color management.
+        case VideoColorSpace::MatrixID::BT709:
+        case VideoColorSpace::MatrixID::UNSPECIFIED:
+        case VideoColorSpace::MatrixID::BT470BG:
+        case VideoColorSpace::MatrixID::SMPTE170M:
+        case VideoColorSpace::MatrixID::BT2020_NCL:
+          break;
+
+          // Supported with color management.
+        case VideoColorSpace::MatrixID::RGB:
+        case VideoColorSpace::MatrixID::FCC:
+        case VideoColorSpace::MatrixID::SMPTE240M:
+        case VideoColorSpace::MatrixID::YCOCG:
+        case VideoColorSpace::MatrixID::YDZDX:
+        case VideoColorSpace::MatrixID::BT2020_CL:
+          break;
+
+          // Never supported.
+        case VideoColorSpace::MatrixID::INVALID:
+          return false;
+      }
+
+      if (color_space.range() == gfx::ColorSpace::RangeID::INVALID) {
+        return false;
+      }
 
-template <typename T>
-class SupplementalProfileCache {
- public:
-  void UpdateCache(const base::flat_set<T>& profiles) {
-    base::AutoLock lock(profiles_lock_);
-    profiles_ = profiles;
-  }
-  bool IsProfileSupported(T profile) {
-    base::AutoLock lock(profiles_lock_);
-    return profiles_.find(profile) != profiles_.end();
-  }
-
- private:
-  base::Lock profiles_lock_;
-  base::flat_set<T> profiles_ GUARDED_BY(profiles_lock_);
-};
-
-SupplementalProfileCache<VideoCodecProfile>*
-GetSupplementalDecoderVideoProfileCache() {
-  static base::NoDestructor<SupplementalProfileCache<VideoCodecProfile>> cache;
-  return cache.get();
-}
-
-SupplementalProfileCache<AudioType>* GetSupplementalDecoderAudioTypeCache() {
-  static base::NoDestructor<SupplementalProfileCache<AudioType>> cache;
-  return cache.get();
-}
-
-SupplementalProfileCache<VideoCodecProfile>*
-GetSupplementalEncoderVideoProfileCache() {
-  static base::NoDestructor<SupplementalProfileCache<VideoCodecProfile>> cache;
-  return cache.get();
-}
-
-bool IsDecoderSupportedHdrMetadata(const VideoType& type) {
-  switch (type.hdr_metadata_type) {
-    case gfx::HdrMetadataType::kNone:
       return true;
-
-    case gfx::HdrMetadataType::kSmpteSt2086:
-      // HDR metadata is currently only used with the PQ transfer function.
-      // See gfx::ColorTransform for more details.
-      return type.color_space.transfer() ==
-             VideoColorSpace::TransferID::SMPTEST2084;
-
-    // 2094-10 SEI metadata is not the same as Dolby Vision RPU metadata, Dolby
-    // Vision decoders on each platform only support Dolby Vision RPU metadata.
-    case gfx::HdrMetadataType::kSmpteSt2094_10:
-    case gfx::HdrMetadataType::kSmpteSt2094_40:
+    }
+
+    #if !BUILDFLAG(USE_PROPRIETARY_CODECS)
+    bool IsVideoCodecProprietary(VideoCodec codec) {
+      switch (codec) {
+        case VideoCodec::kVC1:
+        case VideoCodec::kH264:
+        case VideoCodec::kMPEG2:
+        case VideoCodec::kMPEG4:
+        case VideoCodec::kHEVC:
+        case VideoCodec::kDolbyVision:
+          return true;
+        case VideoCodec::kUnknown:
+        case VideoCodec::kTheora:
+        case VideoCodec::kVP8:
+        case VideoCodec::kVP9:
+        case VideoCodec::kAV1:
+          return false;
+      }
+    }
+
+    bool IsAudioCodecProprietary(AudioCodec codec) {
+      switch (codec) {
+        case AudioCodec::kAAC:
+        case AudioCodec::kAC3:
+        case AudioCodec::kEAC3:
+        case AudioCodec::kAMR_NB:
+        case AudioCodec::kAMR_WB:
+        case AudioCodec::kGSM_MS:
+        case AudioCodec::kALAC:
+        case AudioCodec::kMpegHAudio:
+        case AudioCodec::kDTS:
+        case AudioCodec::kDTSXP2:
+        case AudioCodec::kDTSE:
+        case AudioCodec::kAC4:
+          return true;
+
+        case AudioCodec::kFLAC:
+        case AudioCodec::kIAMF:
+        case AudioCodec::kMP3:
+        case AudioCodec::kOpus:
+        case AudioCodec::kVorbis:
+        case AudioCodec::kPCM:
+        case AudioCodec::kPCM_MULAW:
+        case AudioCodec::kPCM_S16BE:
+        case AudioCodec::kPCM_S24BE:
+        case AudioCodec::kPCM_ALAW:
+        case AudioCodec::kUnknown:
+          return false;
+      }
+    }
+    #endif  // !BUILDFLAG(USE_PROPRIETARY_CODECS)
+
+    bool IsDecoderHevcProfileSupported(const VideoType& type) {
+      if (!IsColorSpaceSupported(type.color_space)) {
+        return false;
+      }
+
+      #if BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS)
+      return true;
+      #else
+      return GetSupplementalDecoderVideoProfileCache()->IsProfileSupported(
+        type.profile);
+      #endif  // BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS)
+    }
+
+    bool IsDecoderVp9ProfileSupported(const VideoType& type) {
+      #if BUILDFLAG(IS_ANDROID)
+      // After Q, all VP9 profiles are required by Android
+      return IsColorSpaceSupported(type.color_space);
+      #elif BUILDFLAG(ENABLE_LIBVPX)
+      // High bit depth capabilities may be toggled via LibVPX config flags.
+      static const bool vpx_supports_hbd = (vpx_codec_get_caps(vpx_codec_vp9_dx()) &
+      VPX_CODEC_CAP_HIGHBITDEPTH) != 0;
+
+      // Color management required for HDR to not look terrible.
+      if (!IsColorSpaceSupported(type.color_space)) {
+        return false;
+      }
+
+      switch (type.profile) {
+        // LibVPX always supports Profiles 0 and 1.
+        case VP9PROFILE_PROFILE0:
+        case VP9PROFILE_PROFILE1:
+          return true;
+        case VP9PROFILE_PROFILE2:
+        case VP9PROFILE_PROFILE3:
+          return vpx_supports_hbd;
+        default:
+          NOTREACHED();
+      }
+      #else
       return false;
-  }
-}
-
-bool IsColorSpaceSupported(const VideoColorSpace& color_space) {
-  switch (color_space.primaries()) {
-    // Transfers supported before color management.
-    case VideoColorSpace::PrimaryID::BT709:
-    case VideoColorSpace::PrimaryID::UNSPECIFIED:
-    case VideoColorSpace::PrimaryID::BT470M:
-    case VideoColorSpace::PrimaryID::BT470BG:
-    case VideoColorSpace::PrimaryID::SMPTE170M:
-      break;
-
-    // Supported with color management.
-    case VideoColorSpace::PrimaryID::SMPTE240M:
-    case VideoColorSpace::PrimaryID::FILM:
-    case VideoColorSpace::PrimaryID::BT2020:
-    case VideoColorSpace::PrimaryID::SMPTEST428_1:
-    case VideoColorSpace::PrimaryID::SMPTEST431_2:
-    case VideoColorSpace::PrimaryID::SMPTEST432_1:
-    case VideoColorSpace::PrimaryID::EBU_3213_E:
-      break;
-
-    case VideoColorSpace::PrimaryID::INVALID:
+      #endif  // BUILDFLAG(ENABLE_LIBVPX)
+    }
+
+    bool IsDecoderAV1Supported(const VideoType& type) {
+      // If the AV1 decoder is enabled, or if we're on Q or later, yes.
+      #if BUILDFLAG(ENABLE_AV1_DECODER)
+      return IsColorSpaceSupported(type.color_space);
+      #elif BUILDFLAG(IS_ANDROID)
+      return base::android::android_info::sdk_int() >=
+      base::android::android_info::SDK_VERSION_Q &&
+      IsColorSpaceSupported(type.color_space);
+      #else
       return false;
-  }
-
-  switch (color_space.transfer()) {
-    // Transfers supported before color management.
-    case VideoColorSpace::TransferID::UNSPECIFIED:
-    case VideoColorSpace::TransferID::GAMMA22:
-    case VideoColorSpace::TransferID::BT709:
-    case VideoColorSpace::TransferID::SMPTE170M:
-    case VideoColorSpace::TransferID::BT2020_10:
-    case VideoColorSpace::TransferID::BT2020_12:
-    case VideoColorSpace::TransferID::IEC61966_2_1:
-      break;
-
-    // Supported with color management.
-    case VideoColorSpace::TransferID::GAMMA28:
-    case VideoColorSpace::TransferID::SMPTE240M:
-    case VideoColorSpace::TransferID::LINEAR:
-    case VideoColorSpace::TransferID::LOG:
-    case VideoColorSpace::TransferID::LOG_SQRT:
-    case VideoColorSpace::TransferID::BT1361_ECG:
-    case VideoColorSpace::TransferID::SMPTEST2084:
-    case VideoColorSpace::TransferID::IEC61966_2_4:
-    case VideoColorSpace::TransferID::SMPTEST428_1:
-    case VideoColorSpace::TransferID::ARIB_STD_B67:
-      break;
-
-    // Never supported.
-    case VideoColorSpace::TransferID::INVALID:
+      #endif
+    }
+
+    bool IsDecoderAACSupported(const AudioType& type) {
+      if (type.profile != AudioCodecProfile::kXHE_AAC) {
+        return true;
+      }
+      #if BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && \
+      (BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN))
+      return GetSupplementalDecoderAudioTypeCache()->IsProfileSupported(type);
+      #else
       return false;
-  }
-
-  switch (color_space.matrix()) {
-    // Supported before color management.
-    case VideoColorSpace::MatrixID::BT709:
-    case VideoColorSpace::MatrixID::UNSPECIFIED:
-    case VideoColorSpace::MatrixID::BT470BG:
-    case VideoColorSpace::MatrixID::SMPTE170M:
-    case VideoColorSpace::MatrixID::BT2020_NCL:
-      break;
-
-    // Supported with color management.
-    case VideoColorSpace::MatrixID::RGB:
-    case VideoColorSpace::MatrixID::FCC:
-    case VideoColorSpace::MatrixID::SMPTE240M:
-    case VideoColorSpace::MatrixID::YCOCG:
-    case VideoColorSpace::MatrixID::YDZDX:
-    case VideoColorSpace::MatrixID::BT2020_CL:
-      break;
-
-    // Never supported.
-    case VideoColorSpace::MatrixID::INVALID:
+      #endif
+    }
+
+    bool IsDecoderDolbyVisionProfileSupported(const VideoType& type) {
+      #if BUILDFLAG(ENABLE_PLATFORM_HEVC) &&                        \
+      BUILDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_DECODE_SUPPORT) && \
+      BUILDFLAG(ENABLE_PLATFORM_DOLBY_VISION)
+      return GetSupplementalDecoderVideoProfileCache()->IsProfileSupported(
+        type.profile);
+      #else
       return false;
-  }
-
-  if (color_space.range() == gfx::ColorSpace::RangeID::INVALID) {
-    return false;
-  }
-
-  return true;
-}
-
-#if !BUILDFLAG(USE_PROPRIETARY_CODECS)
-bool IsVideoCodecProprietary(VideoCodec codec) {
-  switch (codec) {
-    case VideoCodec::kVC1:
-    case VideoCodec::kH264:
-    case VideoCodec::kMPEG2:
-    case VideoCodec::kMPEG4:
-    case VideoCodec::kHEVC:
-    case VideoCodec::kDolbyVision:
+      #endif
+    }
+
+    bool IsDecoderDolbyAc3Eac3Supported(const AudioType& type) {
+      #if BUILDFLAG(ENABLE_PLATFORM_AC3_EAC3_AUDIO)
+      #if BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && \
+      (BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC))
+      return GetSupplementalDecoderAudioTypeCache()->IsProfileSupported(type);
+      #else
+      // Keep 'true' for other platforms as old code snippet.
       return true;
-    case VideoCodec::kUnknown:
-    case VideoCodec::kTheora:
-    case VideoCodec::kVP8:
-    case VideoCodec::kVP9:
-    case VideoCodec::kAV1:
+      #endif  // BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && (BUILDFLAG(IS_WIN) ||
+      // BUILDFLAG(IS_MAC))
+      #else
       return false;
-  }
-}
-
-bool IsAudioCodecProprietary(AudioCodec codec) {
-  switch (codec) {
-    case AudioCodec::kAAC:
-    case AudioCodec::kAC3:
-    case AudioCodec::kEAC3:
-    case AudioCodec::kAMR_NB:
-    case AudioCodec::kAMR_WB:
-    case AudioCodec::kGSM_MS:
-    case AudioCodec::kALAC:
-    case AudioCodec::kMpegHAudio:
-    case AudioCodec::kDTS:
-    case AudioCodec::kDTSXP2:
-    case AudioCodec::kDTSE:
-    case AudioCodec::kAC4:
-      return true;
-
-    case AudioCodec::kFLAC:
-    case AudioCodec::kIAMF:
-    case AudioCodec::kMP3:
-    case AudioCodec::kOpus:
-    case AudioCodec::kVorbis:
-    case AudioCodec::kPCM:
-    case AudioCodec::kPCM_MULAW:
-    case AudioCodec::kPCM_S16BE:
-    case AudioCodec::kPCM_S24BE:
-    case AudioCodec::kPCM_ALAW:
-    case AudioCodec::kUnknown:
+      #endif  // BUILDFLAG(ENABLE_PLATFORM_AC3_EAC3_AUDIO)
+    }
+
+    bool IsDecoderDolbyAc4Supported(const AudioType& type) {
+      #if BUILDFLAG(ENABLE_PLATFORM_AC4_AUDIO) && \
+      BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && BUILDFLAG(IS_WIN)
+      return GetSupplementalDecoderAudioTypeCache()->IsProfileSupported(type);
+      #else
       return false;
-  }
-}
-#endif  // !BUILDFLAG(USE_PROPRIETARY_CODECS)
+      #endif  // BUILDFLAG(ENABLE_PLATFORM_AC4_AUDIO) &&
+      // BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && BUILDFLAG(IS_WIN)
+    }
+
+    bool IsEncoderH264BuiltInVideoType(const VideoType& type) {
+      if (!IsOpenH264SoftwareEncoderEnabled()) {
+        return false;
+      }
+
+      switch (type.profile) {
+        case H264PROFILE_BASELINE:
+        case H264PROFILE_MAIN:
+        case H264PROFILE_HIGH:
+        case H264PROFILE_EXTENDED:
+          return true;
+        case H264PROFILE_HIGH10PROFILE:
+        case H264PROFILE_HIGH422PROFILE:
+        case H264PROFILE_HIGH444PREDICTIVEPROFILE:
+        case H264PROFILE_SCALABLEBASELINE:
+        case H264PROFILE_SCALABLEHIGH:
+        case H264PROFILE_STEREOHIGH:
+        case H264PROFILE_MULTIVIEWHIGH:
+          // Although some of these profiles are supported by openH264, but we don't
+          // wire them for now.
+          return false;
+        default:
+          NOTREACHED();
+      }
+    }
+
+    bool IsEncoderVp8BuiltInVideoType(const VideoType& type) {
+      return BUILDFLAG(ENABLE_LIBVPX);
+    }
+
+    bool IsEncoderVp9BuiltInVideoType(const VideoType& type) {
+      #if BUILDFLAG(ENABLE_LIBVPX)
+      // High bit depth capabilities may be toggled via LibVPX config flags.
+      static const bool vpx_supports_hbd = (vpx_codec_get_caps(vpx_codec_vp9_dx()) &
+      VPX_CODEC_CAP_HIGHBITDEPTH) != 0;
+
+      switch (type.profile) {
+        // LibVPX always supports Profiles 0 and 1.
+        case VP9PROFILE_PROFILE0:
+        case VP9PROFILE_PROFILE1:
+          return true;
+        case VP9PROFILE_PROFILE2:
+        case VP9PROFILE_PROFILE3:
+          return vpx_supports_hbd;
+        default:
+          NOTREACHED();
+      }
+      #else
+      return false;
+      #endif  // BUILDFLAG(ENABLE_LIBVPX)
+    }
+
+    bool IsEncoderAv1BuiltInVideoType(const VideoType& type) {
+      #if BUILDFLAG(ENABLE_LIBAOM)
+      switch (type.profile) {
+        case AV1PROFILE_PROFILE_MAIN:
+        case AV1PROFILE_PROFILE_HIGH:
+          return true;
+        case AV1PROFILE_PROFILE_PRO:
+          // We don't build libaom with high bit depth support.
+          return false;
+        default:
+          NOTREACHED();
+      }
+      #else
+      return false;
+      #endif  // BUILDFLAG(ENABLE_LIBAOM)
+    }
 
-bool IsDecoderHevcProfileSupported(const VideoType& type) {
-  if (!IsColorSpaceSupported(type.color_space)) {
-    return false;
-  }
+  }  // namespace
 
-#if BUILDFLAG(ENABLE_PLATFORM_HEVC)
-#if BUILDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_DECODE_SUPPORT)
-#if BUILDFLAG(IS_CHROMEOS)
-  if (!base::FeatureList::IsEnabled(kPlatformHEVCDecoderSupport)) {
-    return false;
-  }
-#endif  // BUILDFLAG(IS_CHROMEOS)
-  return GetSupplementalDecoderVideoProfileCache()->IsProfileSupported(
-      type.profile);
-#else
-  return true;
-#endif  // BUIDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_DECODE_SUPPORT)
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_PLATFORM_HEVC)
-}
-
-bool IsDecoderVp9ProfileSupported(const VideoType& type) {
-#if BUILDFLAG(IS_ANDROID)
-  // After Q, all VP9 profiles are required by Android
-  return IsColorSpaceSupported(type.color_space);
-#elif BUILDFLAG(ENABLE_LIBVPX)
-  // High bit depth capabilities may be toggled via LibVPX config flags.
-  static const bool vpx_supports_hbd = (vpx_codec_get_caps(vpx_codec_vp9_dx()) &
-                                        VPX_CODEC_CAP_HIGHBITDEPTH) != 0;
-
-  // Color management required for HDR to not look terrible.
-  if (!IsColorSpaceSupported(type.color_space)) {
-    return false;
+  bool IsDecoderSupportedAudioType(const AudioType& type) {
+    if (auto* media_client = GetMediaClient())
+      return media_client->IsDecoderSupportedAudioType(type);
+    return IsDefaultDecoderSupportedAudioType(type);
   }
 
-  switch (type.profile) {
-    // LibVPX always supports Profiles 0 and 1.
-    case VP9PROFILE_PROFILE0:
-    case VP9PROFILE_PROFILE1:
-      return true;
-    case VP9PROFILE_PROFILE2:
-    case VP9PROFILE_PROFILE3:
-      return vpx_supports_hbd;
-    default:
-      NOTREACHED();
+  bool IsDecoderSupportedVideoType(const VideoType& type) {
+    if (auto* media_client = GetMediaClient())
+      return media_client->IsDecoderSupportedVideoType(type);
+    return IsDefaultDecoderSupportedVideoType(type);
   }
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_LIBVPX)
-}
-
-bool IsDecoderAV1Supported(const VideoType& type) {
-  // If the AV1 decoder is enabled, or if we're on Q or later, yes.
-#if BUILDFLAG(ENABLE_AV1_DECODER)
-  return IsColorSpaceSupported(type.color_space);
-#elif BUILDFLAG(IS_ANDROID)
-  return base::android::android_info::sdk_int() >=
-             base::android::android_info::SDK_VERSION_Q &&
-         IsColorSpaceSupported(type.color_space);
-#else
-  return false;
-#endif
-}
 
-bool IsDecoderAACSupported(const AudioType& type) {
-  if (type.profile != AudioCodecProfile::kXHE_AAC) {
-    return true;
-  }
-#if BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && \
-    (BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN))
-  return GetSupplementalDecoderAudioTypeCache()->IsProfileSupported(type);
-#else
-  return false;
-#endif
-}
-
-bool IsDecoderDolbyVisionProfileSupported(const VideoType& type) {
-#if BUILDFLAG(ENABLE_PLATFORM_HEVC) &&                      \
-    BUILDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_DECODE_SUPPORT) && \
-    BUILDFLAG(ENABLE_PLATFORM_DOLBY_VISION)
-  return GetSupplementalDecoderVideoProfileCache()->IsProfileSupported(
-      type.profile);
-#else
-  return false;
-#endif
-}
-
-bool IsDecoderDolbyAc3Eac3Supported(const AudioType& type) {
-#if BUILDFLAG(ENABLE_PLATFORM_AC3_EAC3_AUDIO)
-#if BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && \
-    (BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC))
-  return GetSupplementalDecoderAudioTypeCache()->IsProfileSupported(type);
-#else
-  // Keep 'true' for other platforms as old code snippet.
-  return true;
-#endif  // BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && (BUILDFLAG(IS_WIN) ||
-        // BUILDFLAG(IS_MAC))
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_PLATFORM_AC3_EAC3_AUDIO)
-}
-
-bool IsDecoderDolbyAc4Supported(const AudioType& type) {
-#if BUILDFLAG(ENABLE_PLATFORM_AC4_AUDIO) && \
-    BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && BUILDFLAG(IS_WIN)
-  return GetSupplementalDecoderAudioTypeCache()->IsProfileSupported(type);
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_PLATFORM_AC4_AUDIO) &&
-        // BUILDFLAG(ENABLE_MOJO_AUDIO_DECODER) && BUILDFLAG(IS_WIN)
-}
-
-bool IsEncoderH264BuiltInVideoType(const VideoType& type) {
-  if (!IsOpenH264SoftwareEncoderEnabled()) {
-    return false;
+  bool IsEncoderSupportedVideoType(const VideoType& type) {
+    if (auto* media_client = GetMediaClient()) {
+      return media_client->IsEncoderSupportedVideoType(type);
+    }
+    return IsDefaultEncoderSupportedVideoType(type);
   }
 
-  switch (type.profile) {
-    case H264PROFILE_BASELINE:
-    case H264PROFILE_MAIN:
-    case H264PROFILE_HIGH:
-    case H264PROFILE_EXTENDED:
-      return true;
-    case H264PROFILE_HIGH10PROFILE:
-    case H264PROFILE_HIGH422PROFILE:
-    case H264PROFILE_HIGH444PREDICTIVEPROFILE:
-    case H264PROFILE_SCALABLEBASELINE:
-    case H264PROFILE_SCALABLEHIGH:
-    case H264PROFILE_STEREOHIGH:
-    case H264PROFILE_MULTIVIEWHIGH:
-      // Although some of these profiles are supported by openH264, but we don't
-      // wire them for now.
+  // TODO(chcunningham): Add platform specific logic for Android (move from
+  // MimeUtilInternal).
+  bool IsDefaultDecoderSupportedVideoType(const VideoType& type) {
+    if (!IsDecoderSupportedHdrMetadata(type)) {
       return false;
-    default:
-      NOTREACHED();
-  }
-}
+    }
 
-bool IsEncoderVp8BuiltInVideoType(const VideoType& type) {
-  return BUILDFLAG(ENABLE_LIBVPX);
-}
-
-bool IsEncoderVp9BuiltInVideoType(const VideoType& type) {
-#if BUILDFLAG(ENABLE_LIBVPX)
-  // High bit depth capabilities may be toggled via LibVPX config flags.
-  static const bool vpx_supports_hbd = (vpx_codec_get_caps(vpx_codec_vp9_dx()) &
-                                        VPX_CODEC_CAP_HIGHBITDEPTH) != 0;
-
-  switch (type.profile) {
-    // LibVPX always supports Profiles 0 and 1.
-    case VP9PROFILE_PROFILE0:
-    case VP9PROFILE_PROFILE1:
-      return true;
-    case VP9PROFILE_PROFILE2:
-    case VP9PROFILE_PROFILE3:
-      return vpx_supports_hbd;
-    default:
-      NOTREACHED();
-  }
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_LIBVPX)
-}
-
-bool IsEncoderAv1BuiltInVideoType(const VideoType& type) {
-#if BUILDFLAG(ENABLE_LIBAOM)
-  switch (type.profile) {
-    case AV1PROFILE_PROFILE_MAIN:
-    case AV1PROFILE_PROFILE_HIGH:
-      return true;
-    case AV1PROFILE_PROFILE_PRO:
-      // We don't build libaom with high bit depth support.
+    #if !BUILDFLAG(USE_PROPRIETARY_CODECS)
+    if (IsVideoCodecProprietary(type.codec))
       return false;
-    default:
-      NOTREACHED();
+    #endif
+
+    switch (type.codec) {
+      case VideoCodec::kTheora:
+        return IsDecoderBuiltInVideoCodec(type.codec);
+      case VideoCodec::kH264:
+        return true;
+      case VideoCodec::kVP8:
+        return IsDecoderBuiltInVideoCodec(type.codec)
+        ? true
+        : GetSupplementalDecoderVideoProfileCache()
+        ->IsProfileSupported(type.profile);
+      case VideoCodec::kAV1:
+        return IsDecoderAV1Supported(type);
+      case VideoCodec::kVP9:
+        return IsDecoderVp9ProfileSupported(type);
+      case VideoCodec::kHEVC:
+        return IsDecoderHevcProfileSupported(type);
+      case VideoCodec::kDolbyVision:
+        return IsDecoderDolbyVisionProfileSupported(type);
+      case VideoCodec::kUnknown:
+      case VideoCodec::kVC1:
+      case VideoCodec::kMPEG2:
+      case VideoCodec::kMPEG4:
+        return false;
+    }
   }
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_LIBAOM)
-}
-
-}  // namespace
-
-bool IsDecoderSupportedAudioType(const AudioType& type) {
-  if (auto* media_client = GetMediaClient())
-    return media_client->IsDecoderSupportedAudioType(type);
-  return IsDefaultDecoderSupportedAudioType(type);
-}
-
-bool IsDecoderSupportedVideoType(const VideoType& type) {
-  if (auto* media_client = GetMediaClient())
-    return media_client->IsDecoderSupportedVideoType(type);
-  return IsDefaultDecoderSupportedVideoType(type);
-}
-
-bool IsEncoderSupportedVideoType(const VideoType& type) {
-  if (auto* media_client = GetMediaClient()) {
-    return media_client->IsEncoderSupportedVideoType(type);
-  }
-  return IsDefaultEncoderSupportedVideoType(type);
-}
 
-// TODO(chcunningham): Add platform specific logic for Android (move from
-// MimeUtilInternal).
-bool IsDefaultDecoderSupportedVideoType(const VideoType& type) {
-  if (!IsDecoderSupportedHdrMetadata(type)) {
-    return false;
-  }
-
-#if !BUILDFLAG(USE_PROPRIETARY_CODECS)
-  if (IsVideoCodecProprietary(type.codec))
-    return false;
-#endif
+  bool IsDefaultDecoderSupportedAudioType(const AudioType& type) {
+    if (type.spatial_rendering)
+      return false;
 
-  switch (type.codec) {
-    case VideoCodec::kTheora:
-      return IsDecoderBuiltInVideoCodec(type.codec);
-    case VideoCodec::kH264:
-      return true;
-    case VideoCodec::kVP8:
-      return IsDecoderBuiltInVideoCodec(type.codec)
-                 ? true
-                 : GetSupplementalDecoderVideoProfileCache()
-                       ->IsProfileSupported(type.profile);
-    case VideoCodec::kAV1:
-      return IsDecoderAV1Supported(type);
-    case VideoCodec::kVP9:
-      return IsDecoderVp9ProfileSupported(type);
-    case VideoCodec::kHEVC:
-      return IsDecoderHevcProfileSupported(type);
-    case VideoCodec::kDolbyVision:
-      return IsDecoderDolbyVisionProfileSupported(type);
-    case VideoCodec::kUnknown:
-    case VideoCodec::kVC1:
-    case VideoCodec::kMPEG2:
-    case VideoCodec::kMPEG4:
+    #if !BUILDFLAG(USE_PROPRIETARY_CODECS)
+    if (IsAudioCodecProprietary(type.codec))
       return false;
+    #endif
+
+    switch (type.codec) {
+      case AudioCodec::kAAC:
+        return IsDecoderAACSupported(type);
+      case AudioCodec::kFLAC:
+      case AudioCodec::kMP3:
+      case AudioCodec::kOpus:
+      case AudioCodec::kPCM:
+      case AudioCodec::kPCM_MULAW:
+      case AudioCodec::kPCM_S16BE:
+      case AudioCodec::kPCM_S24BE:
+      case AudioCodec::kPCM_ALAW:
+      case AudioCodec::kVorbis:
+        return true;
+      case AudioCodec::kAMR_NB:
+      case AudioCodec::kAMR_WB:
+      case AudioCodec::kGSM_MS:
+      case AudioCodec::kALAC:
+      case AudioCodec::kMpegHAudio:
+      case AudioCodec::kIAMF:
+      case AudioCodec::kUnknown:
+        return false;
+      case AudioCodec::kDTS:
+      case AudioCodec::kDTSXP2:
+      case AudioCodec::kDTSE:
+        return BUILDFLAG(ENABLE_PLATFORM_DTS_AUDIO);
+      case AudioCodec::kAC3:
+      case AudioCodec::kEAC3:
+        return IsDecoderDolbyAc3Eac3Supported(type);
+      case AudioCodec::kAC4:
+        return IsDecoderDolbyAc4Supported(type);
+    }
   }
-}
-
-bool IsDefaultDecoderSupportedAudioType(const AudioType& type) {
-  if (type.spatial_rendering)
-    return false;
 
-#if !BUILDFLAG(USE_PROPRIETARY_CODECS)
-  if (IsAudioCodecProprietary(type.codec))
-    return false;
-#endif
+  bool IsDefaultEncoderSupportedVideoType(const VideoType& type) {
+    #if !BUILDFLAG(USE_PROPRIETARY_CODECS)
+    if (IsVideoCodecProprietary(type.codec)) {
+      return false;
+    }
+    #endif
 
-  switch (type.codec) {
-    case AudioCodec::kAAC:
-      return IsDecoderAACSupported(type);
-    case AudioCodec::kFLAC:
-    case AudioCodec::kMP3:
-    case AudioCodec::kOpus:
-    case AudioCodec::kPCM:
-    case AudioCodec::kPCM_MULAW:
-    case AudioCodec::kPCM_S16BE:
-    case AudioCodec::kPCM_S24BE:
-    case AudioCodec::kPCM_ALAW:
-    case AudioCodec::kVorbis:
+    if (IsEncoderBuiltInVideoType(type)) {
       return true;
-    case AudioCodec::kAMR_NB:
-    case AudioCodec::kAMR_WB:
-    case AudioCodec::kGSM_MS:
-    case AudioCodec::kALAC:
-    case AudioCodec::kMpegHAudio:
-    case AudioCodec::kIAMF:
-    case AudioCodec::kUnknown:
-      return false;
-    case AudioCodec::kDTS:
-    case AudioCodec::kDTSXP2:
-    case AudioCodec::kDTSE:
-      return BUILDFLAG(ENABLE_PLATFORM_DTS_AUDIO);
-    case AudioCodec::kAC3:
-    case AudioCodec::kEAC3:
-      return IsDecoderDolbyAc3Eac3Supported(type);
-    case AudioCodec::kAC4:
-      return IsDecoderDolbyAc4Supported(type);
-  }
-}
+    }
 
-bool IsDefaultEncoderSupportedVideoType(const VideoType& type) {
-#if !BUILDFLAG(USE_PROPRIETARY_CODECS)
-  if (IsVideoCodecProprietary(type.codec)) {
-    return false;
-  }
-#endif
+    if (IsEncoderOptionalVideoType(type)) {
+      return GetSupplementalEncoderVideoProfileCache()->IsProfileSupported(
+        type.profile);
+    }
 
-  if (IsEncoderBuiltInVideoType(type)) {
-    return true;
+    return false;
   }
 
-  if (IsEncoderOptionalVideoType(type)) {
-    return GetSupplementalEncoderVideoProfileCache()->IsProfileSupported(
-        type.profile);
+  bool IsEncoderBuiltInVideoType(const VideoType& type) {
+    switch (type.codec) {
+      case VideoCodec::kH264:
+        return IsEncoderH264BuiltInVideoType(type);
+      case VideoCodec::kVP8:
+        return IsEncoderVp8BuiltInVideoType(type);
+      case VideoCodec::kAV1:
+        return IsEncoderAv1BuiltInVideoType(type);
+      case VideoCodec::kVP9:
+        return IsEncoderVp9BuiltInVideoType(type);
+      case VideoCodec::kHEVC:
+      case VideoCodec::kTheora:
+      case VideoCodec::kDolbyVision:
+      case VideoCodec::kUnknown:
+      case VideoCodec::kVC1:
+      case VideoCodec::kMPEG2:
+      case VideoCodec::kMPEG4:
+        return false;
+    }
   }
 
-  return false;
-}
-
-bool IsEncoderBuiltInVideoType(const VideoType& type) {
-  switch (type.codec) {
-    case VideoCodec::kH264:
-      return IsEncoderH264BuiltInVideoType(type);
-    case VideoCodec::kVP8:
-      return IsEncoderVp8BuiltInVideoType(type);
-    case VideoCodec::kAV1:
-      return IsEncoderAv1BuiltInVideoType(type);
-    case VideoCodec::kVP9:
-      return IsEncoderVp9BuiltInVideoType(type);
-    case VideoCodec::kHEVC:
-    case VideoCodec::kTheora:
-    case VideoCodec::kDolbyVision:
-    case VideoCodec::kUnknown:
-    case VideoCodec::kVC1:
-    case VideoCodec::kMPEG2:
-    case VideoCodec::kMPEG4:
+  bool IsEncoderOptionalVideoType(const media::VideoType& type) {
+    if (IsEncoderBuiltInVideoType(type)) {
       return false;
+    }
+    switch (type.codec) {
+      case media::VideoCodec::kHEVC:
+        // HEVC only has platform encoder support.
+        return BUILDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_ENCODE_SUPPORT);
+      case media::VideoCodec::kH264:
+      case media::VideoCodec::kAV1:
+      case media::VideoCodec::kVP8:
+      case media::VideoCodec::kVP9:
+        // Check the optional video type when the requested encoding profile does
+        // not match the built‑in type, or when the built‑in encoder is not
+        // available.
+        return true;
+      case media::VideoCodec::kTheora:
+      case media::VideoCodec::kDolbyVision:
+      case media::VideoCodec::kUnknown:
+      case media::VideoCodec::kVC1:
+      case media::VideoCodec::kMPEG2:
+      case media::VideoCodec::kMPEG4:
+        return false;
+    }
   }
-}
 
-bool IsEncoderOptionalVideoType(const media::VideoType& type) {
-  if (IsEncoderBuiltInVideoType(type)) {
-    return false;
-  }
-  switch (type.codec) {
-    case media::VideoCodec::kHEVC:
-      // HEVC only has platform encoder support.
-      return BUILDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_ENCODE_SUPPORT);
-    case media::VideoCodec::kH264:
-    case media::VideoCodec::kAV1:
-    case media::VideoCodec::kVP8:
-    case media::VideoCodec::kVP9:
-      // Check the optional video type when the requested encoding profile does
-      // not match the built‑in type, or when the built‑in encoder is not
-      // available.
+  bool IsDecoderBuiltInVideoCodec(VideoCodec codec) {
+    #if BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS) && BUILDFLAG(USE_PROPRIETARY_CODECS)
+    if (codec == VideoCodec::kH264 || codec == VideoCodec::kHEVC) {
       return true;
-    case media::VideoCodec::kTheora:
-    case media::VideoCodec::kDolbyVision:
-    case media::VideoCodec::kUnknown:
-    case media::VideoCodec::kVC1:
-    case media::VideoCodec::kMPEG2:
-    case media::VideoCodec::kMPEG4:
-      return false;
+    }
+    #endif  // BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS) &&
+    // BUILDFLAG(USE_PROPRIETARY_CODECS)
+    #if BUILDFLAG(ENABLE_LIBVPX)
+    if (codec == VideoCodec::kVP8 || codec == VideoCodec::kVP9) {
+      return true;
+    }
+    #endif  // BUILDFLAG(ENABLE_LIBVPX)
+    #if BUILDFLAG(ENABLE_AV1_DECODER)
+    if (codec == VideoCodec::kAV1)
+      return true;
+    #endif  // BUILDFLAG(ENABLE_AV1_DECODER)
+    return false;
   }
-}
 
-bool IsDecoderBuiltInVideoCodec(VideoCodec codec) {
-#if BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS) && BUILDFLAG(USE_PROPRIETARY_CODECS)
-  if (codec == VideoCodec::kH264) {
-    return true;
+  bool MayHaveAndAllowSelectOSSoftwareEncoder(VideoCodec codec) {
+    // Allow OS software encoding when we don't have an equivalent
+    // software encoder.
+    const bool kHasBundledH264Encoder = IsOpenH264SoftwareEncoderEnabled();
+    constexpr bool kHasOSSoftwareH264Encoder =
+    BUILDFLAG(IS_MAC) || BUILDFLAG(IS_ANDROID);
+    constexpr bool kHasOSSoftwareHEVCEncoder =
+    BUILDFLAG(IS_MAC) && BUILDFLAG(ENABLE_HEVC_PARSER_AND_HW_DECODER);
+
+    switch (codec) {
+      case VideoCodec::kH264:
+        // Prefer the bundled encoder, if present.
+        return kHasOSSoftwareH264Encoder && !kHasBundledH264Encoder;
+      case VideoCodec::kHEVC:
+        return kHasOSSoftwareHEVCEncoder;
+      default:
+        return false;
+    }
   }
-#endif  // BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS) &&
-        // BUILDFLAG(USE_PROPRIETARY_CODECS)
-#if BUILDFLAG(ENABLE_LIBVPX)
-  if (codec == VideoCodec::kVP8 || codec == VideoCodec::kVP9) {
-    return true;
-  }
-#endif  // BUILDFLAG(ENABLE_LIBVPX)
-#if BUILDFLAG(ENABLE_AV1_DECODER)
-  if (codec == VideoCodec::kAV1)
-    return true;
-#endif  // BUILDFLAG(ENABLE_AV1_DECODER)
-  return false;
-}
-
-bool MayHaveAndAllowSelectOSSoftwareEncoder(VideoCodec codec) {
-  // Allow OS software encoding when we don't have an equivalent
-  // software encoder.
-  const bool kHasBundledH264Encoder = IsOpenH264SoftwareEncoderEnabled();
-  constexpr bool kHasOSSoftwareH264Encoder =
-      BUILDFLAG(IS_MAC) || BUILDFLAG(IS_ANDROID);
-  constexpr bool kHasOSSoftwareHEVCEncoder =
-      BUILDFLAG(IS_MAC) && BUILDFLAG(ENABLE_HEVC_PARSER_AND_HW_DECODER);
-
-  switch (codec) {
-    case VideoCodec::kH264:
-      // Prefer the bundled encoder, if present.
-      return kHasOSSoftwareH264Encoder && !kHasBundledH264Encoder;
-    case VideoCodec::kHEVC:
-      return kHasOSSoftwareHEVCEncoder;
-    default:
-      return false;
-  }
-}
-
-void UpdateDefaultDecoderSupportedVideoProfiles(
-    const base::flat_set<media::VideoCodecProfile>& profiles) {
-  GetSupplementalDecoderVideoProfileCache()->UpdateCache(profiles);
-}
-
-void UpdateDefaultDecoderSupportedAudioTypes(
-    const base::flat_set<AudioType>& types) {
-  GetSupplementalDecoderAudioTypeCache()->UpdateCache(types);
-}
 
-void UpdateDefaultEncoderSupportedVideoProfiles(
+  void UpdateDefaultDecoderSupportedVideoProfiles(
     const base::flat_set<media::VideoCodecProfile>& profiles) {
-  GetSupplementalEncoderVideoProfileCache()->UpdateCache(profiles);
-}
+    GetSupplementalDecoderVideoProfileCache()->UpdateCache(profiles);
+    }
+
+    void UpdateDefaultDecoderSupportedAudioTypes(
+      const base::flat_set<AudioType>& types) {
+      GetSupplementalDecoderAudioTypeCache()->UpdateCache(types);
+      }
+
+      void UpdateDefaultEncoderSupportedVideoProfiles(
+        const base::flat_set<media::VideoCodecProfile>& profiles) {
+        GetSupplementalEncoderVideoProfileCache()->UpdateCache(profiles);
+        }
 
 }  // namespace media
diff --git a/media/filters/ffmpeg_demuxer.cc b/media/filters/ffmpeg_demuxer.cc
index 45d82a351a1bc..48be2a4268269 100644
--- a/media/filters/ffmpeg_demuxer.cc
+++ b/media/filters/ffmpeg_demuxer.cc
@@ -57,6 +57,13 @@
 #include "media/filters/ffmpeg_h265_to_annex_b_bitstream_converter.h"
 #endif
 
+#ifndef AV_PKT_DATA_DYNAMIC_HDR_SMPTE_2094_APP5
+#define AV_PKT_DATA_DYNAMIC_HDR_SMPTE_2094_APP5 100 // Safe dummy value
+struct AVDynamicHDRSmpte2094App5 {
+  uint8_t data[1]; // Minimum size
+};
+#endif
+
 namespace media {
 
 namespace {
@@ -343,23 +350,8 @@ base::span<const uint8_t> GetMatroskaBlockAdditionalSideData(
 
 std::vector<uint8_t> GetAgtmSideData(const AVPacket* packet) {
   std::vector<uint8_t> data;
-  size_t side_data_size = 0;
-  AVDynamicHDRSmpte2094App5* side_data =
-      reinterpret_cast<AVDynamicHDRSmpte2094App5*>(av_packet_get_side_data(
-          packet, AV_PKT_DATA_DYNAMIC_HDR_SMPTE_2094_APP5, &side_data_size));
-  if (side_data == nullptr || side_data_size == 0) {
-    return data;
-  }
-  size_t size = 0;
-  if (av_dynamic_hdr_smpte2094_app5_to_t35(side_data, nullptr, &size) != 0) {
-    return data;
-  }
-  data.resize(size);
-  uint8_t* vector_data = data.data();
-  if (av_dynamic_hdr_smpte2094_app5_to_t35(side_data, &vector_data, &size) !=
-      0) {
-    data.clear();
-  }
+  // We return empty 'data' immediately because our FFmpeg
+  // doesn't support the SMPTE 2094 App5 metadata structures.
   return data;
 }
 
diff --git a/media/filters/ffmpeg_video_decoder.cc b/media/filters/ffmpeg_video_decoder.cc
index eaf9f1c3c4490..1c913ed333cda 100644
--- a/media/filters/ffmpeg_video_decoder.cc
+++ b/media/filters/ffmpeg_video_decoder.cc
@@ -73,7 +73,6 @@ static int GetFFmpegVideoDecoderThreadCount(const VideoDecoderConfig& config) {
     case VideoCodec::kUnknown:
     case VideoCodec::kVC1:
     case VideoCodec::kMPEG2:
-    case VideoCodec::kHEVC:
     case VideoCodec::kVP9:
     case VideoCodec::kAV1:
     case VideoCodec::kDolbyVision:
@@ -84,6 +83,7 @@ static int GetFFmpegVideoDecoderThreadCount(const VideoDecoderConfig& config) {
       NOTREACHED();
 
     case VideoCodec::kH264:
+    case VideoCodec::kHEVC:
       // Normalize to three threads for 1080p content, then scale linearly
       // with number of pixels.
       // Examples:
@@ -117,8 +117,9 @@ static void ReleaseVideoBufferImpl(void* opaque, uint8_t* data) {
 
 // static
 bool FFmpegVideoDecoder::IsCodecSupported(VideoCodec codec) {
-  // We only build support for H.264.
-  return codec == VideoCodec::kH264 && IsDecoderBuiltInVideoCodec(codec);
+   // We only build support for H.264 / H.265.
+   return (codec == VideoCodec::kH264 || codec == VideoCodec::kHEVC) &&
+          IsDecoderBuiltInVideoCodec(codec);
 }
 
 FFmpegVideoDecoder::FFmpegVideoDecoder(std::unique_ptr<MediaLog> media_log)
diff --git a/media/webrtc/webrtc_features.cc b/media/webrtc/webrtc_features.cc
index 8e650078a5816..6614714c08c78 100644
--- a/media/webrtc/webrtc_features.cc
+++ b/media/webrtc/webrtc_features.cc
@@ -59,12 +59,12 @@ BASE_FEATURE(kWebRtcAllowH265Receive, base::FEATURE_ENABLED_BY_DEFAULT);
 
 // When enabled, allows H.265 L1T2 to be used for sending WebRTC streams, if the
 // accelerator reports support of encoding in L1T2.
-BASE_FEATURE(kWebRtcH265L1T2, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kWebRtcH265L1T2, base::FEATURE_ENABLED_BY_DEFAULT);
 
 // When enabled, allows H.265 L1T3 to be used for sending WebRTC streams, if the
 // accelerator reports support of encoding in L1T3. If this is enabled, L1T2 is
 // also implied to be enabled.
-BASE_FEATURE(kWebRtcH265L1T3, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kWebRtcH265L1T3, base::FEATURE_ENABLED_BY_DEFAULT);
 
 // When enabled, allows AV1 HW encoding to be used for WebRTC streams, if the
 // platform accelerator supports encoding of AV1.
diff --git a/ui/ozone/platform/wayland/gpu/wayland_surface_factory.cc b/ui/ozone/platform/wayland/gpu/wayland_surface_factory.cc
index 2605b0e267a0e..8bf1a184040d2 100644
--- a/ui/ozone/platform/wayland/gpu/wayland_surface_factory.cc
+++ b/ui/ozone/platform/wayland/gpu/wayland_surface_factory.cc
@@ -246,9 +246,9 @@ GLOzone* WaylandSurfaceFactory::GetGLOzone(
 std::unique_ptr<gpu::VulkanImplementation>
 WaylandSurfaceFactory::CreateVulkanImplementation(bool use_swiftshader,
                                                   bool allow_protected_memory) {
-  LOG_IF(ERROR, !use_swiftshader)
-      << "'--ozone-platform=wayland' is not compatible with Vulkan. "
-         "Consider switching to '--ozone-platform=x11' or disabling Vulkan";
+  // LOG_IF(ERROR, !use_swiftshader)
+  //     << "'--ozone-platform=wayland' is not compatible with Vulkan. "
+  //        "Consider switching to '--ozone-platform=x11' or disabling Vulkan";
   return std::make_unique<VulkanImplementationWayland>(use_swiftshader);
 }
 #endif
